src/dotnet-format.csproj

```diff
@@ -54,6 +54,12 @@
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" />
     <PackageReference Include="Microsoft.Extensions.FileSystemGlobbing" />
     <PackageReference Include="Microsoft.Extensions.Logging" />
+
+    <!--
+      Explicit System.Text.Json package reference is required for source-build to pick up the live package.
+      Avoids picking up an old version via transitive dependency from Microsoft.Build or Microsoft.CodeAnalysis.Workspaces.MSBuild.
+    -->
+    <PackageReference Include="System.Text.Json"  Condition=" '$(DotnetBuildFromSource)' == 'true'" />
```

Comment:
Is it necessary to make this source-build conditioned?

Reply:
I wanted to limit the impact of this change, in servicing. Should we try to make it applicable for all scenarios?

Reply:
I'll let the repo owners decide.  My general recommendation is to avoid source-build specific conditions as much as possible.

Reply:
> I'll let the repo owners decide. My general recommendation is to avoid source-build specific conditions as much as possible.

Thanks. @JoeRobich any concerns with making this change applicable for all scenarios and not just source-build?

@MichaelSimons , I have the same condition in the `sdk` change: https://github.com/dotnet/sdk/pull/33618

Reply:
@sharwell do you have any concerns with updating this PR to remove the source-build condition and make the new package reference applicable in all build environments?

Reply:
It seems that we don't have a confirmation yet that source-build condition should be removed. Will merge by EOD unless I hear otherwise.

@sharwell the question was about modifying this PR to remove source-build condition - any thoughts?